inference-engine/src/auto_plugin/auto_exec_network.cpp

```diff
@@ -15,40 +15,79 @@
 namespace AutoPlugin {
 using namespace InferenceEngine;
 
-AutoExecutableNetwork::AutoExecutableNetwork(const SoExecutableNetworkInternal& network) :
-    _network(network) {
+AutoExecutableNetwork::AutoExecutableNetwork(NetworkTaskSharedPtr cpuTask,
+                                             NetworkTaskSharedPtr acceleratorTask,
+                                             InferenceEngine::IStreamsExecutor::Ptr cpuExecutor)
+                                             : _cpuExecutor(cpuExecutor) {
+    // we wait for any network to become ready (maybe this will already an actual device)
+    if (cpuTask) {
+        _networkFirstReady = cpuTask->get_future().get();
```

Comment:
I think the Executor lifetime is bound to AutoExecutableNetwork by [_cpuExecutor](https://github.com/openvinotoolkit/openvino/pull/5944/files#diff-750d1c33e67b07ac0387961ac068deeeaa0e041e6d9b6cbdcf9f17bb96151201R21)

Reply:
`cpuTask` task is run in executor that was created in plugin::LoadNetworkImpl. As it is gotten from the executor manager the executor and its threads are bound to executor manager lifetime not to executor network.

Reply:
I understand what you mean, `cpuTask ` shared ptr will run in `IStreamsExecutor` (that created by executor manager), but the shared cpuTask will not expire because it is refered in AutoExecutableNetwork, we could get_future().get() in AutoExecutableNetwork, is that right?
Sorry for the update code since rebasing to master branch (fix conflict) and trying std::async(), so code is changed, sorry for that.